### PR TITLE
Add default spark image

### PR DESCRIPTION
**Pull request by pingsutw**
# TL;DR
New image size 3.2 GB vs [Old image](https://github.com/flyteorg/flytesnacks/blob/master/cookbook/integrations/kubernetes/k8s_spark/Dockerfile) size 5.13 GB

```bash
 24K    ./lib/python3.9/site-packages/pyspark/sql/avro
 12K    ./lib/python3.9/site-packages/pyspark/sql/pandas/_typing/protocols
 24K    ./lib/python3.9/site-packages/pyspark/sql/pandas/_typing
104K    ./lib/python3.9/site-packages/pyspark/sql/pandas/__pycache__
264K    ./lib/python3.9/site-packages/pyspark/sql/pandas
1.6M    ./lib/python3.9/site-packages/pyspark/sql
301M    ./lib/python3.9/site-packages/pyspark

=====================================


```

- Add a docker file in `/flytekitplugins-spark`
- Update GH workflow to build spark image
- Use default executor path and applications path when using default spark image

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [x] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
image

```python
import datetime
import random
from operator import add

import flytekit
from flytekit import Resources, task, workflow
from flytekitplugins.spark import Spark
from flytekit.image_spec.image_spec import ImageSpec

spark_image = ImageSpec(base_image="pingsutw/spark-base:v15",
                        registry="pingsutw")


@task(
    task_config=Spark(
        # this configuration is applied to the spark cluster
        spark_conf={
            "spark.driver.memory": "1000M",
            "spark.executor.memory": "1000M",
            "spark.executor.cores": "1",
            "spark.executor.instances": "2",
            "spark.driver.cores": "1",
        }
    ),
    limits=Resources(mem="2000M"),
    cache_version="1",
    container_image=spark_image,
)
def hello_spark(partitions: int) -> float:
    print("Starting Sparkfk wifth Partitions: {}".format(partitions))
    n = 100000 * partitions
    sess = flytekit.current_context().spark_session
    count = (
        sess.sparkContext.parallelize(range(1, n + 1), partitions).map(f).reduce(add)
    )
    pi_val = 4.0 * count / n
    print("Pi val is :{}".format(pi_val))
    return pi_val


def f(_):
    x = random.random() * 2 - 1
    y = random.random() * 2 - 1
    return 1 if x**2 + y**2 <= 1 else 0


@task(cache_version="1", container_image=spark_image)
def print_every_time(value_to_print: float, date_triggered: datetime.datetime) -> int:
    print("My printed value: {} @ {}".format(value_to_print, date_triggered))
    return 1


@workflow
def wf(triggered_date: datetime.datetime = datetime.datetime.now()) -> float:
    """
    Using the workflow is still as any other workflow. As image is a property of the task, the workflow does not care
    about how the image is configured.
    """
    pi = hello_spark(partitions=50)
    print_every_time(value_to_print=pi, date_triggered=triggered_date)
    return pi


if __name__ == "__main__":
    print(f"Running {__file__} main...")
    print(
        f"Running my_spark(triggered_date=datetime.datetime.now()){wf(triggered_date=datetime.datetime.now())}"
    )

```

## Tracking Issue
_NA_

## Follow-up issue
_NA_

### CODE REVIEW

File: plugins/flytekit-envd/flytekitplugins/envd/image_builder.py [hunk 1]
```diff
@@ -48,16 +48,20 @@ def build():
...
+        envd_config += f'    install.python(version="{image_spec.python_version}")\n'
+
     ctx = context_manager.FlyteContextManager.current_context()
     cfg_path = ctx.file_access.get_random_local_path("build.envd")
     pathlib.Path(cfg_path).parent.mkdir(parents=True, exist_ok=True)
 
     if image_spec.source_root:
         shutil.copytree(image_spec.source_root, pathlib.Path(cfg_path).parent, dirs_exist_ok=True)
+        # Indentation is required by envd
         envd_config += '    io.copy(host_path="./", envd_path="/root")'
```
Comment (eapolinario) [inline plugins/flytekit-envd/flytekitplugins/envd/image_builder.py:65]: nit: should we also add a newline here?

**Review by eapolinario — APPROVED** [review]
one nit, otherwise LGTM. Approving in case you think addressing the nit is not necessary.

File: plugins/flytekit-envd/flytekitplugins/envd/image_builder.py (hunk 1 above)
Comment (pingsutw) [reply plugins/flytekit-envd/flytekitplugins/envd/image_builder.py:65]: not necessary since it's the last line of config
